### PR TITLE
Multiple Changes (2.0.0)

### PR DESCRIPTION
**Pull request by twrodriguez**
- Removed DynamoDB
- Added QueryParams, LookupHash, NestedHash
- Added 'find_by' collection method; alias 'pmap' to 'parallel_map'
- Include RubyDig to polyfill Hash#dig and Array#dig on older rubies
- ThreadPool now uses ConcurrentRuby under the hood for better
  thread-safety. ThreadPool also explicitly kills all threads on join.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,25 @@
 # og-ruby-utils
-Basic utilities and boilerplate code for ruby development
+Basic utilities and boilerplate code for ruby development. Included is #dig support for older
+versions of ruby as well as [Concurrent Ruby](https://github.com/ruby-concurrency/concurrent-ruby)
+
+## Classes
```
Comment (cademarkegard) [inline README.md]: we need to one day like... fill in some examples <3

Comment (hannahsquier) [inline README.md]: that would be helpful :)

File: spec/lib/thread_pool_spec.rb [hunk 2]
```diff
@@ -90,9 +94,12 @@
       expect(returns).to include(0 => 1, 1 => 2, 2 => 3, -1 => 'huzzah')
       expect(elapsed).to be > 0.5
       expect(elapsed).to be < 1.0
+      sleep 0.1
```
Comment (cademarkegard) [inline spec/lib/thread_pool_spec.rb]: not a fan but I guesssssss

File: spec/lib/query_params_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    it 'encodes params that contain arrays' do
+      expect(OpenGov::Util::QueryParams.encode('hello' => %w(yolo1 yolo2))).to eq('?hello%5B%5D=yolo1&hello%5B%5D=yolo2')
+    end
+
+    it 'encodes multiple params' do
+      encoded_params = OpenGov::Util::QueryParams.encode(
+        'hello1' => %w(yolo1 yolo11),
+        :hello2 => 'yolo2',
+        'hello3' => 'yolo3'
```
Comment (cademarkegard) [inline spec/lib/query_params_spec.rb]: YOLO

Comment (cazzer) [inline spec/lib/query_params_spec.rb]: YODO

**Pull request merged**